client/src/main/java/io/confluent/kafka/schemaregistry/client/rest/utils/UrlList.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.confluent.kafka.schemaregistry.client.rest.utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * This class manages a set of urls for accessing an upstream registry. It basically
```

Comment:
Any thoughts on how we're distributing requests and the final failure policy? Trying each server one time is probably a good policy. Are we sure we want to pin to the last known good server? That's good if the reason for having multiple servers is failover, less good if you're trying to load balance between them. I'm not sure the load balancing is a big concern since schema registry requests are cheap and clients ideally only do one at startup, but it'd be good to make sure the approach we take in this class is well justified.